modules/dashboard/php/endpoints/charthelper.class.inc

```diff
@@ -0,0 +1,316 @@
+<?php declare(strict_types=1);
+/**
+ * POST request methods.
+ *
+ * Used with c3.js for the Dashboard.
+ *
+ * PHP Version 7
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+namespace LORIS\dashboard\endpoints;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Api\Endpoint;
+use \LORIS\Middleware\ETagCalculator;
+
+/**
+ * The NDB_Page for the POST requests.
+ *
+ * @category Loris
+ * @package  Dashboard
+ * @author   Tara Campbell <tara.campbell@mail.mcgill.ca>
+ *           Alizée Wickenheiser <alizee.wickenheiser@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class ChartHelper extends Endpoint implements ETagCalculator
+{
+    public $skipTemplate = true;
+    /**
+     * This function will return a json response.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Ensure GET or POST request.
+        switch ($request->getMethod()) {
+        case 'GET':
+            return $this->_handleGET($request);
+        default:
+            return new \LORIS\Http\Response\MethodNotAllowed(
+                $this->allowedMethods()
+            );
+        }
+    }
+
+    /**
+     * Initialize setup, the extra values for the
+     * create timepoint form. (psc & errors)
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request.
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handleGET(ServerRequestInterface $request) : ResponseInterface
+    {
+        // Parse GET query params.
+        $values   = $request->getQueryParams();
+        $response = array();
+        if (isset($values['get'])
```

Comment:
Would be better to check for `!isset()` once rather than do it on every branch.

See https://en.wikipedia.org/wiki/Happy_path